discord/guild.py

```diff
@@ -353,6 +354,19 @@ def _voice_state_for(self, user_id: int, /) -> Optional[VoiceState]:
     def _add_member(self, member: Member, /) -> None:
         self._members[member.id] = member
 
+    def _update_member(self, payload: MemberPayload, user_id: int, /) -> Member:
+        if user_id in self._members:
+            member = self.get_member(user_id)
+            member._update(payload)
+        else:
+            # NOTE:
+            # This is a fallback in case the member is not found in the guild's members.
+            # If this fallback occurs, multiple aspects of the Member
+            # class will be incorrect such as status and activities.
+            member = Member(guild=self, state=self._state, data=payload)  # type: ignore
+            self._members[user_id] = member
```

Comment:
Should there be a new `MemberCacheFlags` flag for this?

Reply:
As in adding a flag to MemberCacheFlags to mark if the data is potentially invalid or not?